webapp/packages/plugin-sql-editor/src/SqlResultTabs/OutputLogs/useOutputLogsPanelState.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2023 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { action, computed, observable } from 'mobx';
+
+import { useObservableRef } from '@cloudbeaver/core-blocks';
+import type { WsOutputLogInfo } from '@cloudbeaver/core-sdk';
+
+import type { ISqlEditorTabState } from '../../ISqlEditorTabState';
+
+export interface SqlOutputLogsPanelState {
+  searchValue: string;
+  setSearchValue: (value: string) => void;
+  logMessages: WsOutputLogInfo['message'][];
+  selectedLogTypes: IOutputLogType[];
+  readonly resultValue: string;
+  setSelectedLogTypes: (value: IOutputLogType[]) => void;
+  readonly filteredLogs: WsOutputLogInfo[];
+}
+
+export const OUTPUT_LOG_TYPES = ['Debug', 'Log', 'Info', 'Notice', 'Warning', 'Error'] as const;
+export type IOutputLogType = (typeof OUTPUT_LOG_TYPES)[number];
+
+export const useOutputLogsPanelState = (outputLogs: WsOutputLogInfo[], sqlEditorTabState: ISqlEditorTabState) =>
+  useObservableRef<SqlOutputLogsPanelState>(
+    () => ({
+      searchValue: '',
+      get selectedLogTypes() {
+        return sqlEditorTabState.outputLogsTab?.selectedLogTypes || [];
+      },
+      setSearchValue(value: string) {
+        this.searchValue = value;
+      },
+      setSelectedLogTypes(value: IOutputLogType[]) {
+        if (sqlEditorTabState.outputLogsTab) {
+          sqlEditorTabState.outputLogsTab.selectedLogTypes = value;
+        }
+      },
```

Comment:
please move it to `OutputLogsService.ts`. So logic will be next:
you have `IOutputLogsTab` state
you have `OutputLogsService`, and you place logic to manage state here. Those functions can accept the whole SQL editor state or just `IOutputLogsTab`

You have menu actions and handlers. These handlers need to access the SQL editor state or `IOutputLogsTab` so they can use `OutputLogsService` to manage the state.

Sometimes we can use an abstract interface that can have implementation as react state/object, but this case is bad practice in general and usually needs to reflect some actions from services on UI; for example `TreeNodeMenu` places `INodeActions` to context and then from action handler this interface uses to open an inline editor for node rename